### ISSUE TITLE
UX issues

### ISSUE THREAD

**Issue by colllin** [auth0/lock#393]
Have you guys done any user testing with the Login / Signup toggle?  We're seeing huge issues there with knowing which one is selected, even with the heading directly above.

Another issue we're seeing is not recognizing the submit button as a button.

We're planning to write custom CSS to try to fix these issues.  Would you be interested in us submitting those changes as a PR if we see success?  

Here's a good estimate of what we're going to try:
![image](https://cloud.githubusercontent.com/assets/185923/15330537/8295da58-1c1a-11e6-9612-901363e8767a.png)

Summary of changes:

- Tabs instead of a toggle make it more clear which one is selected.
- Heading is app name instead of "Log In" or "Sign Up".
- Submit button has padding.
- Submit button has text that reinforces the selected tab.

**Comment by dleeward** [comment]
We had same issues In our initial testing. If you don't share CSS on GitHub, please share with us.

**Comment by rickyrauch** [comment]
@colllin thanks for the feedback, we are going to work on new concepts for this tab component this week.

**Comment by colllin** [comment]
Did you guys come up with any new concepts?  If not, we're going to implement the design I pasted above.

Here's another issue we're finding, this time one that I can't fix with CSS:
![image](https://cloud.githubusercontent.com/assets/185923/15544145/ee065534-2254-11e6-95d0-8041e443be62.png)

It seems that when a Rule redirects to a different tab, it's not getting focus / the user isn't seeing it.  Do you think the opened tab could pass some data back to the Lock module so that it could display some text, maybe that we can customize.  e.g. we would like this loading icon to instead say "Please complete your registration in the popup window".  It's not really correct to show a loading icon here because that indicates to the user that the page is doing something useful and they should wait.  In fact, the page is waiting for the *user* to manually complete the process.  I think it's actually possible to put a link that would trigger the opened window using `<a target="window_name">`.

How does an Auth0 Rule redirect behave if the user has popups blocked in their browser settings?

**Comment by gnandretta** [comment]
Hi @colllin, we are still ironing out some details but this is what it will look like.

<img width="305" alt="screen shot 2016-06-01 at 4 07 37 pm" src="https://cloud.githubusercontent.com/assets/120195/15722256/33dad062-2813-11e6-970e-bf539ca16cdb.png">

With regards to the second question, can you provide more detailed steps to reproduce the issue? please include how you are initializing Lock. 

Thank you.

**Comment by colllin** [comment]
@gnandretta 😍  Looks great!  `Auth0` heading is customizable too, or is that part of the logo?

I think this will definitely solve the biggest problem – not knowing which tab is selected.  If we try additional ideas like ensuring that the submit button looks like a button and is clear about what it does, I'll let you know if it has affect on user completion.

Re: the second question.  I'm not sure that it's something to "reproduce".  We have a Rule that redirects to a registration form.  Based on video recordings of users' screens, it seems that they don't notice the popup containing the registration form.  We're using basic lock functionality: `lock = new Auth0Lock(clientId, domain); lock.show({icon: ..., primaryColor: ...});`.  After thinking about it more, I think the loading icon is wrong in this case.  A loading icon signals to the user that the interface is working in the background, and the user should wait.  In this case, the interface is actually waiting on the user, which is the default state of an interface.  I think it would be ideal if, instead of a loading icon, we put a helpful message and a link that focuses the popup window (`<a target="window_name">`).

**Comment by dleeward** [comment]
I still don't agree that it is unambiguous. The way that "Signup" is grayed out, it doesn't appear to be clickable. I personally like @collin 's mockup better. I also like text in addition to the > sign.

**Comment by gnandretta** [comment]
@colllin the heading is customizable too.

When Lock opens the popup it shows the loading indicator and yes that it confusing, we will fix it. But I'm not sure I fully understand your problem. If you show lock with

```js
lock.show({icon: ..., primaryColor: ...});
```

it will use redirect mode and won't open a popup. Also the popup window normally opens in the middle of the screen, I am curious about how a user may miss that.

Can you share a screenshot of Lock and the popup window open?

**Comment by colllin** [comment]
@gnandretta We're passing a callback... sorry, I shouldn't have left that part out.  Does that change the behavior?  Now that I remove the callback and set `responseType: 'code'`, I see the redirect to our registration form and back, and I think this is exactly what we want for now.  Thank you.

**Comment by colllin** [comment]
This is getting off-topic, but it's not clear how to use the `?code=abc123` that comes back from this flow...

**Comment by colllin** [comment]
Docs for using `?code=abc123` are [here under **Regular Webapp**](https://auth0.com/docs/libraries/lock/types-of-applications#redirect-mode).

Removing the callback from `lock.show()` but not setting `responseType: 'code'` works as expected – the response token is parsed by `lock.parseHash()`.

**Comment by gnandretta** [comment]
The callback activates popup mode. If possible you should omit it and use redirect mode because it provides a better user experience.

The `code` you were receiving is intended to be used by the server. It can be exchanged for a token. Because you are handling auth in the browser you want a `responseType: "token"` (which in your case is the [default](https://auth0.com/docs/libraries/lock/customization#responsetype-string-)).

We made some changes to the API of [next version of Lock](https://auth0.com/docs/libraries/lock/v10). We tried to make it easier and to provide better defaults.

**Comment by hzalaz** [comment]
Closing this since the underlying ux issue was fixed. Please go to https://support.auth0.com to ask for support.
